### PR TITLE
Build: update all `build.tools` and add latest versions

### LINKED ISSUE #11350: Build tools: update `latest` versions
## What's the problem this feature will solve?

The version we have of miniconda (4.7) is quite old, I remember checking for this some time ago, and thinking that there weren't more versions available (5.x for example). Turns out that they changed their versioning scheme some time ago... Their versions now appear to be linked to a python version and the miniconda version (e.g `miniconda3-3.12-24.1.2.0`).

## Describe the solution you'd like

We are close to June, that's our 6 months policy update for all build tools, so we can maybe wait until that to update it. Or just do one exception, as our policy mentions.

## Alternative solutions

Maybe find a way to manually upgrade miniconda for users that need it?

## Additional context

We have one user not being able to build their docs, a probable cause is the version of miniconda. See https://github.com/numba/pixie/pull/12.

### PR DESCRIPTION
**Pull request by humitos**
Matches https://github.com/readthedocs/readthedocs-docker-images/pull/205

Closes #11350

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11386.org.readthedocs.build/en/11386/

- Developer's documentation (`dev`): https://dev--11386.org.readthedocs.build/en/11386/

### CODE REVIEW

File: readthedocs/builds/constants_docker.py [hunk 1]
```diff
@@ -32,38 +32,42 @@
...
-            "3.10": "3.10.13",
-            "3.11": "3.11.6",
-            "3.12": "3.12.0",
+            "3.8": "3.8.19",
+            "3.9": "3.9.19",
+            "3.10": "3.10.14",
+            "3.11": "3.11.9",
+            "3.12": "3.12.3",
             "miniconda3-4.7": "miniconda3-4.7.12",
+            "miniconda3-3.12": "miniconda3-3.12-24.1.2-0",
```
Comment (stsewd) [inline readthedocs/builds/constants_docker.py:41]: This should include the 24.x in the name, 3.12 references the python version, not the conda version.

**Review by stsewd — APPROVED** [review]

**Comment by simondeziel** [comment]
@humitos many thanks for that! I just tried it with golang 1.22 but it was too soon. Do you have a rough ETA on this for public consumption? Just trying to gauge if I wait a few hours or a few days.

**Comment by humitos** [comment]
@simondeziel we are going to deploy these changes next Tuesday. I will suggest you to come back and try this next Wednesday to be sure it will work.

**Comment by simondeziel** [comment]
@humitos excellent, thanks again

**Pull request merged**